centreon/src/Core/ResourceAccess/Application/UseCase/UpdateRule/UpdateRuleRequest.php

```diff
@@ -0,0 +1,52 @@
+<?php
+
+/*
+ * Copyright 2005 - 2023 Centreon (https://www.centreon.com/)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * For more information : contact@centreon.com
+ *
+ */
+
+declare(strict_types=1);
+
+namespace Core\ResourceAccess\Application\UseCase\UpdateRule;
+
+final class UpdateRuleRequest
+{
+    public int $id = 0;
+
+    public string $name = '';
+
+    public string $description = '';
```

Comment:
```suggestion
    public ?string $description = null;
```

In the validation Schema `description` is not in the required fields

Reply:
Even if it is not mandatory I've choosen to keep it not nullable and get a empty string instead. But there is a missing part indeed in the Controller to say "there is no description provided then set an empty value on the UpdateRequest"